From 973c5e43ccbbfde6e982b6cb30b406bd04617560 Mon Sep 17 00:00:00 2001
From: Sami Shalayel <sami.shalayel@qt.io>
Date: Mon, 6 Oct 2025 15:07:56 +0200
Subject: [PATCH 157/239] tst_styleimports: make attachedTypesAvailable use
 builtins styles

Reset the style to "" to use the builtins styles, instead of reusing the
last style set by a previous test. Before this change, `tst_styleimports
attachedTypesAvailable` would test Basic, Fusion, ... styles while
`tst_styleimports` would only test the `FileSystemStyle` style set by a
previous test.
This commit fixes a test failure in a later commit.

Pick-to: 6.8
Task-number: QTBUG-133755
Change-Id: I693b976e786dce4bcf45dcf31b55142aed342a70
Reviewed-by: Mitch Curtis <mitch.curtis@qt.io>
(cherry picked from commit 952452c7555be85af36020390468dc314d168aae)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 tests/auto/quickcontrols/styleimports/tst_styleimports.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/tests/auto/quickcontrols/styleimports/tst_styleimports.cpp b/tests/auto/quickcontrols/styleimports/tst_styleimports.cpp
index a88c154448..7d2d12dc22 100644
--- a/tests/auto/quickcontrols/styleimports/tst_styleimports.cpp
+++ b/tests/auto/quickcontrols/styleimports/tst_styleimports.cpp
@@ -379,6 +379,8 @@ void tst_StyleImports::attachedTypesAvailable_data()
     // QtQuick.Controls import.
     QTest::newRow("Controls") << "";
 
+    // use builtins style, otherwise testStyles() returns the style last set by a previous test
+    QQuickStyle::setStyle(QLatin1String(""));
     const QStringList styles = testStyles();
     for (const QString &styleImport : styles)
         QTest::newRow(qPrintable(styleImport)) << styleImport;
-- 
2.51.2

